### PR TITLE
Personal user space code and keymaps by ZyBeR

### PR DESCRIPTION
**Pull request by ZyberSE**
My personal user space code and keymaps for some of the boards I own.

## Description
This contains various bits of code in my user space section and several keymaps for different boards. I also fixed a compile warning in knife66/knife66_iso for the new deprecated parameter RGBLIGHT_EFFECT_KNIGHT. No changes outside of that is included in this PR.

My keymaps are made to be used on macOS with the "swansi" keymap and ansi layout. Some of the features are TapDaces to write Swedish accent chars, double tap 4 to take a screenshoot and keep it in the clipboard and some boards have a function to paste "code blocks" to Discord, Slack etc by just tapping a single button which triggers a macro which outputs the required back-ticks and pasts the clipboard content.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/playkbtw/ca66/keymaps/zyber/rules.mk [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+BOOTMAGIC_ENABLE = no
+MOUSEKEY_ENABLE = no	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no		# Console for debug
+COMMAND_ENABLE = no		# Commands for debug and configuration
+SLEEP_LED_ENABLE = no	# Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes	# Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/playkbtw/ca66/keymaps/zyber/rules.mk:26]:
```suggestion
MOUSEKEY_ENABLE = no
```

File: keyboards/wilba_tech/zeal65/rules.mk [hunk 2]
```diff
@@ -20,7 +20,7 @@ COMMAND_ENABLE = no        # Commands for debug and configuration
 NKRO_ENABLE = yes           # Enable N-Key Rollover
 BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
 AUDIO_ENABLE = no           # Audio output
-RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight. 
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
```
Comment (fauxpark) [inline keyboards/wilba_tech/zeal65/rules.mk:23]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight. 
```

File: users/zyber/zyber.c [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	switch(keycode) {
+		case C_BLK:
+			if (record->event.pressed) {
+				send_string("```" SS_DELAY(80) SS_LCMD("v") SS_DELAY(80) "```");
+			}
+			break;
+
+		case CTRL_C_UP:
+			if (record->event.pressed) {
+				register_mods(MOD_BIT(KC_LCTRL));
```
Comment (fauxpark) [inline users/zyber/zyber.c:26]:
```suggestion
				register_mods(MOD_BIT(KC_LEFT_CTRL));
```

File: users/zyber/zyber.c [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+			} else {
+				unregister_mods(MOD_BIT(KC_LCTRL));
```
Comment (fauxpark) [inline users/zyber/zyber.c:28]:
```suggestion
				unregister_mods(MOD_BIT(KC_LEFT_CTRL));
```

File: users/zyber/zyber.c [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+void dance_SSHT_finished(qk_tap_dance_state_t *state, void *user_data) {
+	if (state->count == 2) {
+		register_code(KC_LGUI);
+		register_code(KC_LCTRL);
+		register_code(KC_LSHIFT);
+		register_code(KC_4);
+		unregister_code(KC_LGUI);
+		unregister_code(KC_LCTRL);
+		unregister_code(KC_LSHIFT);
+		unregister_code(KC_4);
```
Comment (fauxpark) [inline users/zyber/zyber.c:74]:
```suggestion
                tap_code16(C(S(G(KC_4))));
```

File: users/zyber/zyber.c [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+	} else {
+		register_code(KC_4);
+		unregister_code(KC_4);
```
Comment (fauxpark) [inline users/zyber/zyber.c:77]:
```suggestion
		tap_code(KC_4);
```

File: users/zyber/zyber.c [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+	}
+}
+void dance_SSHT_reset(qk_tap_dance_state_t *state, void *user_data) {
+	if (state->count == 2) {
+		unregister_code(KC_LGUI);
+		unregister_code(KC_LCTRL);
+		unregister_code(KC_LSHIFT);
+		unregister_code(KC_4);
```
Comment (fauxpark) [inline users/zyber/zyber.c:85]:
```suggestion
		unregister_code16(C(S(G(KC_4))));
```

File: users/zyber/zyber.c [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+	}
+}
+
+/* Å */
+void dance_LBRC_finished(qk_tap_dance_state_t *state, void *user_data) {
+	if (state->count == 2) {
+		register_code(KC_LALT);
+		register_code(KC_LBRC);
+		unregister_code(KC_LALT);
+		unregister_code(KC_LBRC);
```
Comment (fauxpark) [inline users/zyber/zyber.c:97]:
```suggestion
                tap_code16(A(KC_LBRC));
```

File: users/zyber/zyber.c [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+	} else {
+		register_code(KC_LBRC);
+		unregister_code(KC_LBRC);
```
Comment (fauxpark) [inline users/zyber/zyber.c:100]:
```suggestion
		tap_code(KC_LBRC);
```

File: users/zyber/zyber.c [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+	}
+}
+void dance_LBRC_reset(qk_tap_dance_state_t *state, void *user_data) {
+	if (state->count == 2) {
+		unregister_code(KC_LALT);
+		unregister_code(KC_LBRC);
```
Comment (fauxpark) [inline users/zyber/zyber.c:106]:
```suggestion
		unregister_code16(A(KC_LBRC));
```

File: users/zyber/zyber.c [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+	}
+}
+
+/* Ö */
+void dance_SCLN_finished(qk_tap_dance_state_t *state, void *user_data) {
+	if (state->count == 2) {
+		register_code(KC_LALT);
+		register_code(KC_SCLN);
+		unregister_code(KC_LALT);
+		unregister_code(KC_SCLN);
```
Comment (fauxpark) [inline users/zyber/zyber.c:118]:
```suggestion
		tap_code16(A(KC_SCLN));
```

File: users/zyber/zyber.c [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+	} else {
+		register_code(KC_SCLN);
+		unregister_code(KC_SCLN);
```
Comment (fauxpark) [inline users/zyber/zyber.c:121]:
```suggestion
		tap_code(KC_SCLN);
```

File: users/zyber/zyber.c [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+	}
+}
+
+/* Ä */
+void dance_QUOT_finished(qk_tap_dance_state_t *state, void *user_data) {
+	if (state->count == 2) {
+		register_code(KC_LALT);
+		register_code(KC_QUOT);
+		unregister_code(KC_LALT);
+		unregister_code(KC_QUOT);
```
Comment (fauxpark) [inline users/zyber/zyber.c:139]:
```suggestion
		tap_code16(A(KC_QUOT));
```

File: users/zyber/zyber.c [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+	} else {
+		register_code(KC_QUOT);
+		unregister_code(KC_QUOT);
```
Comment (fauxpark) [inline users/zyber/zyber.c:142]:
```suggestion
		tap_code(KC_QUOT);
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/wilba_tech/zeal65/rules.mk (hunk 2 above)
Comment (ZyberSE) [reply keyboards/wilba_tech/zeal65/rules.mk:23]: Why leave the trailing white space?

Comment (fauxpark) [reply keyboards/wilba_tech/zeal65/rules.mk:23]: That can be fixed in another PR. Try to keep keyboard changes separate from keymap/userspace changes.

File: users/zyber/zyber.c (hunk 5 above)
Comment (ZyberSE) [reply users/zyber/zyber.c:74]: OMG, thank you I didn't know I could do this :)

File: keyboards/wilba_tech/zeal65/rules.mk (hunk 2 above)
Comment (ZyberSE) [reply keyboards/wilba_tech/zeal65/rules.mk:23]: OK

**Comment by ZyberSE** [comment]
The CI build complains about `misterknife/knife66_iso:zyber` issues is I do not have a keymap in that folder. Under misterknife there are two folders, one `knife66` and one `knife66_iso`. My keymap is in `knife66`.
What can I do to resolve this?

**Comment by tzarc** [comment]
> The CI build complains about `misterknife/knife66_iso:zyber` issues is I do not have a keymap in that folder. Under misterknife there are two folders, one `knife66` and one `knife66_iso`. My keymap is in `knife66`. What can I do to resolve this?

This is now fixed.

The other issue along those lines is that you've made modifications to the `config.h` files for both the knife66 keyboards -- you'll need to revert those changes.

**Comment by ZyberSE** [comment]
> > The CI build complains about `misterknife/knife66_iso:zyber` issues is I do not have a keymap in that folder. Under misterknife there are two folders, one `knife66` and one `knife66_iso`. My keymap is in `knife66`. What can I do to resolve this?
> 
> This is now fixed.
> 
> The other issue along those lines is that you've made modifications to the `config.h` files for both the knife66 keyboards -- you'll need to revert those changes.

Not a problem - I'll revert those, but the original code generates compiler warnings due to the deprecated option I removed.

**Review by fauxpark — APPROVED** [review]

File: keyboards/ai03/lunar/keymaps/zyber/config.h [hunk 15]
```diff
@@ -0,0 +1,5 @@
+/**
```
Comment (drashna) [inline keyboards/ai03/lunar/keymaps/zyber/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Also, you want/need an include guard here (`#pragma once`)

File: keyboards/alps64/keymaps/zyber/keymap.c [hunk 16]
```diff
@@ -0,0 +1,31 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/alps64/keymaps/zyber/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```


Also, if you have `#include QMK_KEYBOARD_H` in your zyber.h file, you could just include that here instead.

File: keyboards/converter/adb_usb/keymaps/zyber/keymap.c [hunk 17]
```diff
@@ -0,0 +1,13 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/converter/adb_usb/keymaps/zyber/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/fc660c/keymaps/zyber/keymap.c [hunk 18]
```diff
@@ -0,0 +1,45 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/fc660c/keymaps/zyber/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/playkbtw/ca66/keymaps/zyber/keymap.c [hunk 19]
```diff
@@ -0,0 +1,19 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/playkbtw/ca66/keymaps/zyber/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/wilba_tech/rama_works_m60_a/keymaps/zyber/config.h [hunk 20]
```diff
@@ -0,0 +1,5 @@
+// USB Device descriptor parameter
+#undef PRODUCT
```
Comment (drashna) [inline keyboards/wilba_tech/rama_works_m60_a/keymaps/zyber/config.h:2]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

And an `#pragma once`

File: keyboards/wilba_tech/rama_works_m60_a/keymaps/zyber/keymap.c [hunk 21]
```diff
@@ -0,0 +1,41 @@
+// M60-A layout
```
Comment (drashna) [inline keyboards/wilba_tech/rama_works_m60_a/keymaps/zyber/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/wilba_tech/zeal65/keymaps/zyber/config.h [hunk 22]
```diff
@@ -0,0 +1,28 @@
+#pragma once
```
Comment (drashna) [inline keyboards/wilba_tech/zeal65/keymaps/zyber/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/wilba_tech/zeal65/keymaps/zyber/keymap.c [hunk 23]
```diff
@@ -0,0 +1,38 @@
+// ZyBeRs layout for Zeal65
```
Comment (drashna) [inline keyboards/wilba_tech/zeal65/keymaps/zyber/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: users/zyber/config.h [hunk 24]
```diff
@@ -0,0 +1,8 @@
+#define GRAVE_ESC_GUI_OVERRIDE
```
Comment (drashna) [inline users/zyber/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

and `#pragma once`

File: users/zyber/zyber.c [hunk 25]
```diff
@@ -0,0 +1,134 @@
+#include "zyber.h"
```
Comment (drashna) [inline users/zyber/zyber.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: users/zyber/zyber.h [hunk 26]
```diff
@@ -0,0 +1,29 @@
+#pragma once
```
Comment (drashna) [inline users/zyber/zyber.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request merged**